src/network.lisp

```diff
@@ -23,36 +23,36 @@
   (:report (lambda (condition stream)
              (format stream "Request error: ~A" (what condition)))))
 
+(defun process-options (&rest options &key (pathname-p nil) &allow-other-keys)
+  (loop for (key value)
+          on (alexandria:remove-from-plist options :timeout :streamp :pathname-p)
+	by #'cddr
+	when value
+	  if pathname-p
+	    collect (cons (kebab:to-snake-case key) value)
+	else
```

Comment:
I think here you are breaking the previous logic or this `else` belong to `if` but having a wrong indentation. Seems there is a problem with loop indentation in emacs, because for me in this small example it also indents `else` on the same level as `when`:

```
CL-USER> (loop for i in (list NIL NIL 1 2 3 4 5)
               when i
                 if (< i 3)
                   collect (list :before-else i)
               else
                 collect (list :else i))
((:BEFORE-ELSE 1) (:BEFORE-ELSE 2) (:ELSE 3) (:ELSE 4) (:ELSE 5))
```

Let's rewrite this code to make it more readable?

Also, I don't like an explicit pathname-p argument. We already have a pathname in the options. So we can decide if we wanna to make a multipart request right in the `make-request` itself. I'd add a helper function `(multipart-request-required-p options)` which will return true if at least one of options is `pathname`. Also `pathname-p` argument to `process-options` should be renamed to `return-alist-p` - this way it will be more obvious that this function can return plist or alist depending on this mode.